sched/clustermembership/cluster.go

```diff
@@ -3,8 +3,37 @@ package cluster_membership
 //go:generate mockgen -source=cluster.go -package=cluster_membership -destination=cluster_mock.go
 
 type Cluster interface {
-	/*
-	 * Returns a Slice of Node Ids in the Cluster
-	 */
+	//
+	// Returns a Slice of Node in the Cluster
+	//
 	Members() []Node
 }
+
+type DynamicCluster interface {
```

Comment:
I'd recommend looking at this through the lens of the Interface Segregation Principle, which is the I in SOLID.

I think this is really two interfaces.

The Scheduler wants the updates.

Some other system is what will drive the AddNode and RemoveNode, so that can be a separate interface.